crates/rooch-types/src/transaction/rooch.rs

```diff
@@ -137,7 +135,8 @@ impl AuthenticatableTransaction for RoochTransaction {
     //TODO unify the hash function
     fn tx_hash(&self) -> H256 {
         //TODO cache the hash
-        moveos_types::h256::sha3_256_of(self.encode().as_slice())
+        let data = bcs::to_bytes(&self.data).expect("encode transaction should success");
```

Comment:
There are two hash:

1. the RoochTransaction's hash which includes the signature data and it is the ID of the transaction.
2. the RoochTransactionData's hash which is used for the signature. 

We should put the second one in the RoochTransactionData.